packages/@aws-cdk/aws-eks/lib/service-account.ts

```diff
@@ -78,20 +79,24 @@ export class ServiceAccount extends Construct implements IPrincipal {
     this.grantPrincipal = this.role.grantPrincipal;
     this.policyFragment = this.role.policyFragment;
 
-    cluster.addManifest(`${id}ServiceAccountResource`, {
-      apiVersion: 'v1',
-      kind: 'ServiceAccount',
-      metadata: {
-        name: this.serviceAccountName,
-        namespace: this.serviceAccountNamespace,
-        labels: {
-          'app.kubernetes.io/name': this.serviceAccountName,
+    new KubernetesManifest(this, `manifest-${id}ServiceAccountResource`, {
```

Comment:
add a comment explaining why this cannot use `cluster.addManifest`

Reply:
👍